README.md

```diff
@@ -1,5 +1,232 @@
+
+- [Requirements](#requirements)
+  - [Dependencies](#dependencies)
+- [Getting Started](#getting-started)
+  - [Install the IOTA SDK](#install-the-iota-sdk)
+  - [Usage](#usage)
+    - [Wallet](#wallet)
+    - [Client](#client)
+    - [Examples](#examples)
+  - [API Reference](#api-reference)
+- [Contribute](#contribute)
+- [License](#license)
+- [Acknowledgments](#acknowledgments)
+
+## Features
+
+- **Wallet module**: The `wallet` module in the IOTA SDK provides high-level functions for managing accounts, generating
+  addresses, creating transactions, and interacting with the Shimmer network. It offers a user-friendly interface for
+  developers to build applications on the Shimmer network.
+
+- **Client module**: The `client` module in the IOTA SDK offers low-level functions that allow you to have
+  fine-grained control over their interactions with Shimmer nodes. It provides access to the underlying API endpoints
+  and enables advanced operations such as custom message construction and direct communication with the network.
+
+- **Python, Node.js, and WASM Bindings**: The IOTA SDK includes bindings for Python, Node.js, and WASM, which allow you to use the
+  SDK in your preferred programming language. These bindings provide seamless integration with existing Python and
+  Node.js projects, enabling cross-platform compatibility and flexibility.
+
+## Branching Structure for Development
+
+This library follows the following branching strategy:
+
+| Branch       | Description                                                                                                                    |
+|--------------|--------------------------------------------------------------------------------------------------------------------------------|
+| `develop`    | Ongoing development for future releases of the networks. This branch gets merged into `staging` on releases.                   |
+| `production` | The latest releases for the IOTA network.                                                                                      |
+| `staging`    | The latest releases for the Shimmer network.                                                                                   |
+| other        | Other branches that may reflect current projects. Like `develop`, they will find their way into `staging` once they are ready. |
+
+## Before You Start
+
+This file is focused on Rust. Please refer to the [Python](bindings/python/README.md)
+and [Node.js](bindings/nodejs/README.md) instructions if you want information on installing and using them.
+
+## Requirements
+
+The IOTA SDK requires `Rust` and `Cargo`. You can find installation instructions in
+the [Rust documentation](https://doc.rust-lang.org/cargo/getting-started/installation.html).
+
+We recommend you first update the Rust compiler to the latest stable version:
+
+```shell
+rustup update stable
+```
+
+The nightly Rust compiler should be fine, but some changes might not be compatible.
```

Comment:
this should be elaborated on, or removed

Reply:
```suggestion
```